[Swift] Add functions concept exercise (#2005)
exercism/v3: languages/swift/exercises/concept/functions/README.md

> In this exercise you're going to write some more code to help you cook your brilliant lasagna from your favorite cooking book.

Reviewer: I really like how you've expanded on the `basics` exercise's story in the tasks!

To clean things up, define the `remainingMinutesInOven` function that takes the number of elapsed minutes the lasagna has been in the oven as a parameter, as well as an `expectedMinutesInOven` parameter that has a default value of 40. Your function should return how many minutes the lasagna still has to remain in the oven, based on the expected time oven time in minutes from the previous task.

```swift
remainingMinutesInOven(elapsedMinutes: 30)
// => 10
remainingMinutesInOven(elapsedMinutes: 30, expectedMinutesInOven: 75)
// => 45
```

## 2. Variadic parameters

You had previously written a `preparationTimeInMinutes` function that takes the number of layers you added to the lasagna as a parameter and returns how many minutes you spent preparing the lasagna, assuming each layer takes you 2 minutes to prepare. However, you are not particularly happy with this function, as you have to count the number of layers in your lasagna yourself before you can call this function, and who has time for all that?

> To make things easier for yourself, write a function, `preparationTimeInMinutes` that takes a variadic String parameter, `layers`, where you list the different layers in your lasagna. The function will then count the number of layers in your lasagna, and return the amount of preparation time required.

Reviewer: ```suggestion
To make things easier for yourself, write a function, `preparationTimeInMinutes` that takes a variadic `String` parameter, `layers`, where you list the different layers in your lasagna. The function will then count the number of layers in your lasagna, and return the amount of preparation time required.
```

// => 4
preparationTimeInMinutes(layers: "sauce", "noodles", "sauce", "meat", "mozzarella", "noodles", "sauce", "eggplant", "béchamel", "noodles", "sauce", "mozzarella")
// => 24
```

## 3. Multiple return values

Before you start to assemble your lasagna, you want to make sure you have enough sauce and noodles to build the lasagna of your dreams. For each layer of "noodles" in your lasagna, you will need 3 noodles. For each layer of "sauce" in your lasagna, you will need 0.2 liters of sauce.

Define the function `quantities` that takes a variadic String parameter, `layers`, where you list the different layers in your lasagna. The function will then determine the quantity of noodles and sauce needed to make your meal. The result should be returned in a tuple of type `(noodles: Int, sauce: Double)`.

```swift
quantities(layers: "sauce", "noodles", "sauce", "meat", "mozzarella", "noodles", "sauce", "eggplant", "béchamel", "noodles", "sauce", "mozzarella")
// => (noodles 9, sauce 0.8)
quantities(layers: "sauce", "noodles", "sauce", "noodles", "meat", "noodles", "mozzarella")
// => (noodles 9, sauce 0.4)
```

## 4. In-out parameters

> Unfortunately, the sauce you have on hand was made in America, and the volume is listed in fluid ounces instead of liters. To help let you know if you have enough sauce on hand, write a function, `toOz`, that takes an in-out tuple parameter, `(noodles: Int, sauce: Double)`, and converts the quantity of sauce to fluid ounces. Note that there are 33.814 fluid ounces per liter.

Reviewer: Just to prevent people from reading "Unfortunately, the sauce you have on hand was made in America" as something negative about America:

```suggestion
Unfortunately, the sauce you have on hand has the volume is listed in fluid ounces instead of liters. To help let you know if you have enough sauce on hand, write a function, `toOz`, that takes an in-out tuple parameter, `(noodles: Int, sauce: Double)`, and converts the quantity of sauce to fluid ounces. Note that there are 33.814 fluid ounces per liter.
```
